exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java

```diff
@@ -241,6 +243,30 @@ public void setOption(final OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, OptionType type) {
+    assert type == OptionType.SYSTEM;
+    try { // ensure option exists
```

Comment:
you can extract this try block as a static method `ensureOptionExists`, this same code is duplicated in multiple places